Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project. For complete copyright and license terms please see the LICENSE at the root of
+ * this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Memory/SystemAllocator.h>
+
+#include <AzToolsFramework/FocusMode/FocusModeInterface.h>
+#include <AzToolsFramework/Prefab/PrefabFocusInterface.h>
+#include <AzToolsFramework/Prefab/Template/Template.h>
+
+namespace AzToolsFramework
+{
+    class PrefabEditorEntityOwnershipInterface;
+}
+
+namespace AzToolsFramework::FocusModeFramework
+{
+    class FocusModeInterface;
+}
+
+namespace AzToolsFramework::Prefab
+{
+    class InstanceEntityMapperInterface;
+
+    class PrefabFocusHandler final
```

Comment:
suggestion: API comment for this type could be useful

Reply:
Added!